framework/util/to_string.h

```diff
@@ -68,7 +68,8 @@ template <typename PtrType>
 inline std::string PtrToString(PtrType ptr)
 {
     std::stringstream strStrm;
-    strStrm << "0x" << reinterpret_cast<const void*>(ptr);
+    const void*       val = reinterpret_cast<const void*>(ptr);
```

Comment:
I think on 32-bit `val` may end up being sign-extended, depending on the compiler.  https://godbolt.org/z/v6qro1a3z
Could we `reinterpret_cast` `ptr` directly to `uintptr_t` and then `<<` that?  Like
`strStrm << "0x" << std::hex << reinterpret_cast<uintptr_t>(ptr);`

Reply:
I don't think sign extension would be an issue. None of the inputs, outputs, or intermediate values are signed. A conversion from a pointer to an unsigned 64 bit int shouldn't do sign extension.

Reply:
The godbolt link shows that it does.  I haven't looked into it; I imagine maybe whether or not sign-extension happens may be undefined or something.

Reply:
Oh!  I had a typo in my Compiler Explorer example.  https://godbolt.org/z/7hq55vn56  But the result is the same.

Reply:
Please also change this instance of these two lines to `strStrm << "0x" << std::hex << reinterpret_cast<uintptr_t>(ptr);`